apis/logging/v1/cluster_log_forwarder_types.go

```diff
@@ -148,7 +148,7 @@ type OutputSpec struct {
 
 	// Type of output plugin.
 	//
-	// +kubebuilder:validation:Enum:=syslog;fluentdForward;elasticsearch;kafka;cloudwatch;loki;googleCloudLogging
+	// +kubebuilder:validation:Enum:=syslog;fluentdForward;elasticsearch;kafka;cloudwatch;loki;googleCloudLogging;splunk
```

Comment:
@alanconway someplace in the api comments, I don't know if its here or in the Splunk spec, we should update to state what is supported for which collector:
* fluentd: syslog, forward, es, kafka, loki, cw
* vector: es, kafka, cw, loki, gcp, splunk

Reply:
and a follow up task to where I introduce validation https://github.com/openshift/cluster-logging-operator/pull/1654  we should add validation to reject CLF with reason why

Reply:
+1 How about on the collector field of the CL? "fluentd supports blah..., vector supports blah..." in one place so its easy to compare support and easy to update as we add more support to vector. I doubt we can use // kube: comments to validate this since it involves 2 separate resources (although I haven't taken time to dig into the kube valdation expression language that apparently has appeared.) in which case it'll have to be in code.